mergify_cli/ci/cli.py

```diff
@@ -95,3 +119,83 @@ async def _process_junit_files(  # noqa: PLR0913
             click.style(f"Error uploading JUnit XML reports: {e}", fg="red"),
             err=True,
         )
+
+    failing_spans = [
+        span
+        for span in spans
+        if span.status.status_code == opentelemetry.trace.StatusCode.ERROR
+        and span.attributes is not None
+        and span.attributes.get("test.scope") == "case"
+    ]
+    if not failing_spans:
+        return
+
+    await check_failing_spans_with_quarantine(
+        api_url,
+        token,
+        repository,
+        [fspan.name for fspan in failing_spans],
+    )
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=0.2),
+    stop=tenacity.stop_after_attempt(5),
+    retry=tenacity.retry_if_exception_type(httpx.TransportError),
+    reraise=True,
+)
+async def check_failing_spans_with_quarantine(
+    api_url: str,
+    token: str,
+    repository: str,
+    failing_spans_names: list[str],
+) -> None:
+    fspans_str = "\n".join(failing_spans_names)
+    click.echo(
+        f"Checking the following failing tests for quarantine:\n{fspans_str}",
+        err=False,
+    )
+
+    async with utils.get_http_client(
+        server=f"{api_url}/v1/ci/{repository}/quarantine",
+        headers={"Authorization": f"Bearer {token}"},
+    ) as client:
+        response = await client.post(
+            "/check",
+            json={"tests_names": failing_spans_names},
+        )
+
+        if response.status_code != 200:
+            click.echo(
+                click.style(
+                    f"HTTP error {response.status_code} while checking quarantined tests: {response.text}",
+                    fg="red",
+                ),
+                err=True,
+            )
+            sys.exit(1)
```

Comment:
@Greesb if I read this correctly, it means that in the case that Mergify API fails AND all tests are OK, then we will make the job fail. Is that a good understanding and a good choice?

Reply:
Nope, we only enter this function when there are failed tests in the spans we retrieved.

Reply:
https://github.com/Mergifyio/mergify-cli/pull/713/files#diff-43391424afd9ed6670ca3093c79142f7aa40ead1519d0e3e4d6a58cc3a69eb0bR130